resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
See also the detailed analysis at https://github.com/weaveworks/guides/issues/133#issuecomment-158964627.  Note that if you do some simple tests you may have all addresses with the same prefix, in which case what you said will be true, but once there is a bit more variety the sorting will put some addresses above others.